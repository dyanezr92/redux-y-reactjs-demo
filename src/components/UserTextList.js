import React from 'react'
import PropTypes from 'prop-types'
import List from './List'
import UserText from './UserText'

const UserTextList = ({ elements, elementName, onDeleteClick }) => {

  return (
    <List elements={ elements } emptyMessage={ "No " + elementName + ", dude!" } onDeleteElementClick={ onDeleteClick }>
      { elements.map(
          (userText, i) =>
            <UserText key={userText.id} text={userText.text} author={userText.author} id={userText.id} onDeleteClick={onDeleteClick} />
        )
      }
    </List>
  );
};

UserTextList.propTypes = {
  elements: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string,
    author: PropTypes.string
  }).isRequired).isRequired,
  elementName: PropTypes.string.isRequired,
  onDeleteClick: PropTypes.func.isRequired
}

export default UserTextList
