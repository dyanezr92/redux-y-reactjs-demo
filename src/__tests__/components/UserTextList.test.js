import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

import UserTextList from '../../components/UserTextList';

describe('<UserTextList onUserTextSubmit={ console.log } elementName="comment" />', () => {
  describe('render()', () => {
    test('renders the user text list', () => {
      const wrapper = shallow(
        <UserTextList elements={[ {"id": 1, "author": "Finn", "text": "Awesome comment!"}]} elementName="comments" onDeleteClick={ console.log }/>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
