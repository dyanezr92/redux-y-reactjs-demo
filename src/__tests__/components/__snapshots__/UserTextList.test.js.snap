// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserTextList onUserTextSubmit={ console.log } elementName="comment" /> render() renders the user text list 1`] = `
<List
  elements={
    Array [
      Object {
        "author": "Finn",
        "id": 1,
        "text": "Awesome comment!",
      },
    ]
  }
  emptyMessage="No comments, dude!"
  onDeleteElementClick={[Function]}
>
  <UserText
    author="Finn"
    id={1}
    key="1"
    onDeleteClick={[Function]}
    text="Awesome comment!"
  />
</List>
`;
